File: samples/JavaSamples/src/main/java/com/mku/salmon/samples/main/TextProgram.java
package com.mku.salmon.samples.main;

import com.mku.salmon.samples.samples.SamplesCommon;
import com.mku.salmon.samples.samples.TextSample;
import com.mku.salmon.streams.ProviderType;
import com.mku.salmon.streams.SalmonStream;

import java.io.IOException;

public class TextProgram {
    public static void main(String[] args) throws IOException {
        String password = "test123";
        String text = "This is a plain text that will be encrypted";

        SalmonStream.setAesProviderType(ProviderType.Default);

        // generate an encryption key from the text password
        byte[] key = SamplesCommon.getKeyFromPassword(password);
        System.out.println("Plain Text: " + "\n" + text + "\n");

        String encText = TextSample.encryptText(text, key);
        System.out.println("Encrypted Text: " + "\n" + encText + "\n");

        String decText = TextSample.decryptText(encText, key);
        System.out.println("Decrypted Text: " + "\n" + decText + "\n");
    }
}

File: samples/CSharpSamples/CSharpSamples/Main/TextProgram.cs
using Mku.Salmon.Samples.Samples;
using Mku.Salmon.Streams;

namespace Mku.Salmon.Samples.Main;

class TextProgram
{
    public static void RunMain(string[] args)
    {
        string password = "test123";
        string text = "This is a plain text that will be encrypted";
		
		SalmonStream.AesProviderType = ProviderType.Default;
		
        // generate an encryption key from the text password
        byte[] key = SamplesCommon.GetKeyFromPassword(password);
        Console.WriteLine("Plain Text: " + "\n" + text + "\n");

        string encText = TextSample.EncryptText(text, key);
        Console.WriteLine("Encrypted Text: " + "\n" + encText + "\n");

        string decText = TextSample.DecryptText(encText, key);
        Console.WriteLine("Decrypted Text: " + "\n" + decText + "\n");
    }
}